### PR TITLE
#9 Feat/login form check data input 

### PR DESCRIPTION
**Pull request by minhng271**
### Check data input of form login
- Signup form validate
   - Check empty
   - Check validation
   - Show error message
- Check data user (call API)
   - Check user exists or not  
   - Show error or go to homepage
### Result:
 

https://user-images.githubusercontent.com/49194444/209631698-8ef9f392-90c2-421a-8d0b-b3e7f4d897f4.mp4

### CODE REVIEW

**Comment by minhng271** [comment]
### Update result 

https://user-images.githubusercontent.com/49194444/209744359-9d995bcb-7749-4e19-8688-77533fa8133f.mp4

File: react-basic-practice/src/features/LoginForm/index.jsx [hunk 1]
```diff
@@ -8,30 +8,60 @@ import FormGroup from '../../components/FormGroup';
...
+      setLoading(true);
+
+      // Check validation input
+      const errorValid = validateInput(inputValue);
+
+      // Check data user
+      if (!errorValid.error) {
+        let haveUser = false;
+        const dataUser = await getUserByMail(inputValue.email);
+        dataUser.find((user) => {
```
Comment (MinhNguyen41092) [inline react-basic-practice/src/features/LoginForm/index.jsx]: You can create a function for this piece of logic, so you can reuse it later.

File: react-basic-practice/src/services/Users/index.js [hunk 2]
```diff
@@ -26,7 +26,7 @@ export const getUserById = async (id) => {
  * @returns {object} user
  */
 export const getUserByMail = async (mail) => {
```
Comment (MinhNguyen41092) [inline react-basic-practice/src/services/Users/index.js]: `email` not `mail`

File: react-basic-practice/src/features/LoginForm/index.jsx [hunk 3]
```diff
@@ -8,30 +8,59 @@ import FormGroup from '../../components/FormGroup';
...
 const initialInput = {
   email: '',
-  userName: '',
   password: '',
 };
 
 const LoginForm = () => {
   const [errorMessage, setErrorMessage] = useState(initialErrorMsgs);
   const [inputValue, setInputValue] = useState(initialInput);
+  // const [empty, setEmpty] = useState(false);
```
Comment (MinhNguyen41092) [inline react-basic-practice/src/features/LoginForm/index.jsx]: Remove

**Review by MinhNguyen41092 — APPROVED** [review]

**Pull request merged**